### PR TITLE
[BACKLOG-20780] MQTT Consumer add Quality of Service Level

### PR DESCRIPTION
**Pull request by DFieldFL**

### CODE REVIEW

**Comment by DFieldFL** [comment]
@pentaho/ackbar please review

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumer.java [hunk 1]
```diff
@@ -65,7 +68,14 @@ public boolean init( StepMetaInterface stepMetaInterface, StepDataInterface step
       logger.error( getString( PKG, "MQTTInput.Error.FailureGettingFields" ), e );
     }
     window = new FixedTimeStreamWindow<>( subtransExecutor, rowMeta, getDuration(), getBatchSize() );
-    source = new MQTTStreamSource( mqttConsumerMeta.getMqttServer(), mqttConsumerMeta.getTopics(), 2, this );
+    try {
+      source = new MQTTStreamSource( environmentSubstitute( mqttConsumerMeta.getMqttServer() ),
+        Arrays.asList( environmentSubstitute( mqttConsumerMeta.getTopics().toArray( new String[]{} ) ) ),
```
Comment (kurtwalker) [inline kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumer.java]: should give it an array of the exact size

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumerDialog.java [hunk 2]
```diff
@@ -154,7 +181,7 @@ private void buildTopicsTable( Composite parentWidget, Control controlAbove ) {
     fdData.left = new FormAttachment( 0, 0 );
     fdData.top = new FormAttachment( controlAbove, 5 );
     fdData.right = new FormAttachment( 0, 337 );
-    fdData.bottom = new FormAttachment( 100, 0 );
+    fdData.height = 180;
```
Comment (kurtwalker) [inline kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumerDialog.java]: should be anchored to the control below

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumer.java (hunk 1 above)
Comment (DFieldFL) [reply kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumer.java]: 👍

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumerDialog.java (hunk 2 above)
Comment (DFieldFL) [reply kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumerDialog.java]: 👍

**Pull request merged**